errors/errors.go

```diff
@@ -39,4 +39,4 @@ func (err *QueryError) Error() string {
 	return str
 }
 
-var _ error = &QueryError{}
+var _ error = (*QueryError)(nil)
```

Comment:
I am happy to merge this but it is unrelated change. It needs a separate PR. Let's keep this PR to only breadth first search changes.